### PR TITLE
Separate conversion and validation from iso_format_validation function

### PR DESCRIPTION
**Pull request by josemauro**
iso_format_validation function  were doing validation of the format ISO-8601 and
trying to convert if it wans't in ISO-8601 format. A new function were created called
convert_to_iso.

### CODE REVIEW

File: utils.py [hunk 1]
```diff
@@ -20,15 +20,27 @@ def validate_timestamp(start, end):
     return True
 
 
+def convert_to_iso(timestamp):
+    """Convert a value to to ISO-8601 format."""
```
Comment (hdiogenes) [inline utils.py]:
```suggestion
    """Convert a value to ISO-8601 format."""
```

File: utils.py [hunk 2]
```diff
@@ -20,15 +20,27 @@ def validate_timestamp(start, end):
...
+    try:
+        timestamp = float(timestamp)
+        iso = '%Y-%m-%dT%H:%M:%SZ'
+        timestamp = datetime.utcfromtimestamp(timestamp).strftime(iso)
+        return timestamp
+    except ValueError:
+        log.error(f'Error: Timestamp value \'{timestamp}\' is not at '
```
Comment (hdiogenes) [inline utils.py]:
```suggestion
        log.error(f'Error: Timestamp value \'{timestamp}\' is not in '
```

File: utils.py [hunk 3]
```diff
@@ -41,17 +53,9 @@ def iso_format_validation(timestamp):
...
-            return False
-        except OverflowError:
-            log.error(f'Error: Timestamp \'{timestamp}\' float value is too '
-                      'large to be used as datetime.')
-            return False
-    return timestamp
+    if match_iso(timestamp) is None:
+        return False
+    if match_date(timestamp) is None:
+        return False
```
Comment (hdiogenes) [inline utils.py]:
```
if not (match_iso(timestamp) or match_date(timestamp)):
    return False
```

File: utils.py [hunk 4]
```diff
@@ -20,15 +20,27 @@ def validate_timestamp(start, end):
...
+    except ValueError:
+        log.error(f'Error: Timestamp value \'{timestamp}\' is not in '
+                  'ISO-8601 format.')
+    except OverflowError:
+        log.error(f'Error: Timestamp \'{timestamp}\' float value is too '
+                  'large to be used as datetime.')
+
+
 def iso_format_validation(timestamp):
     """Verify if a timestamp is in isoformat. If it's not try to convert it."""
```
Comment (hdiogenes) [inline utils.py]: Update docstring to reflect changes in code.

File: utils.py [hunk 5]
```diff
@@ -20,15 +20,27 @@ def validate_timestamp(start, end):
...
+        timestamp = float(timestamp)
+        iso = '%Y-%m-%dT%H:%M:%SZ'
+        timestamp = datetime.utcfromtimestamp(timestamp).strftime(iso)
+        return timestamp
+    except ValueError:
+        log.error(f'Error: Timestamp value \'{timestamp}\' is not in '
+                  'ISO-8601 format.')
+    except OverflowError:
+        log.error(f'Error: Timestamp \'{timestamp}\' float value is too '
+                  'large to be used as datetime.')
```
[the remark this answers is not in the document]
Comment (josemauro) [reply utils.py]: Log messages removed and Exceptions are raised.

File: utils.py (hunk 4 above)
Comment (josemauro) [reply utils.py]: Docstring updated.

**Pull request merged**